File: JS Core/2.JS Advanced/12.JS Classes/rectangle.js
class Rectangle {
    constructor(width, height, color){
        this.width = width;
        this.height = height;
        this.color = color;
    }
    calcArea() {
        return this.width * this.height;
     }
}
let rectangle = new Rectangle(4,5,'red');
console.log(rectangle.width);
console.log(rectangle.height);
console.log(rectangle.color);
console.log(rectangle.calcArea());

File: C# Advanced/C# OPP/07.Interfaces and Abstraction/InterfacesAbstractionLab/01.Shapes/Rectangle.cs
using System;

namespace Shapes
{
    public class Rectangle : IDrawable
    {
        private int width;
        private int height;

        public Rectangle(int width, int height)
        {
            this.width = width;
            this.height = height;
        }

        public void Draw()
        {
            FrontPart();
            MiddlePart();
            FrontPart();
        }

        private void MiddlePart()
        {
            for (int row = 0; row < this.height - 2; row++)
            {
                Console.Write("*");
                for (int col = 0; col < this.width - 2; col++)
                {
                    Console.Write(" ");
                }
                Console.WriteLine("*");
            }
        }

        private void FrontPart()
        {
            Console.WriteLine(new string('*', this.width));
        }
    }
}
